qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -79,6 +83,47 @@ def init():
 }
 
 
+def inject_userscripts():
+    """Register user javascript files with the global profiles."""
+    # The greasemonkey metadata block support in qtwebengine only starts at 5.8
+    # Otherwise have to handle injecting the scripts into the page at very
+    # early load, probs same place in view as the enableJS check.
+    if not qtutils.version_check('5.8'):
+        return
+
+    #XXX: Docs say world other than main should also be able to
+    # access the DOM but my scripts aren't working. Needs more testing.
+    userjs_worldid = _JS_WORLD_MAP[usertypes.JsWorld.main]
+
+    # Since we are inserting scripts into profile.scripts they won't
+    # just get replaced by new gm scripts like if we were injecting them
+    # ourselves so we need to remove all gm scripts, while not removing
+    # any other stuff that might have been added. Like the one for
+    # stylsheets.
+    # Could either use a different world for gm scripts, check for gm metadata
+    # values (would mean no non-gm userscripts), or check the code for
+    # _qute_script_id
+    for profile in [webenginesettings.default_profile,
+                    webenginesettings.private_profile]:
+        scripts = profile.scripts()
+        for script in scripts.toList():
+            if script.worldId() == userjs_worldid:
```

Comment:
Why not use `script.setName()` when creating it (and setting that to `GreaseMonkey-<name>` or something), and then checking for that here?